gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityDataServiceEntityFetcher.java

```diff
@@ -84,12 +85,8 @@ public EntityFetcherResponse getEntities(
     }
 
     // Add all expressions in the select that are already not part of the EntityID attributes
-    entitiesRequest.getSelectionList().stream()
-        .filter(expression -> expression.getValueCase() == ValueCase.COLUMNIDENTIFIER)
-        .filter(
-            expression ->
-                !mappedEntityIdAttributeIds.contains(
-                    expression.getColumnIdentifier().getColumnName()))
+    entitiesRequest
```

Comment:
Curious Q: Why did we remove this filter `!mappedEntityIdAttributeIds`? Does this mean we will add those selections twice?

Reply:
The reason it was removed was that we were dropping alias information here - if the caller requested an id attribute by a different alias, the existing implementation was removing that. I had figured that duplicating a selection wouldn't have any significant impact, and wasn't worth the complexity.

That said, in the QS version of the entity fetcher, I believe I ended up with a better solution where I dropped dupe requests and used the aliases at the end to remap them back ( https://github.com/hypertrace/gateway-service/pull/116/files/ea0965d95e98c7caaa83dea5501dc17437362189#diff-11defa5d01d0ff51c801f8ea113e9b9100328b20e0d3073b3faae6bb38c31626R241-R246 ) so let me revisit this and see if that's feasible here.

Reply:
Followed up with the same solution here.